Writerside/topics/raptor.md

```diff
@@ -1,49 +1,61 @@
-it calculates the earliest arrival time at each
-station along the line that has not yet been visited and adds these stations to the list for
-the next round.
-
-**Updating the arrival times**: When the algorithm finds a new stop,
-it updates the earliest arrival time at this stop. If
-the new arrival time is earlier than the previous one, the route is updated.
-
-**Check for termination**: The algorithm ends when it has reached the destination station and
-there are no more stations to
-visit in the current lap, or when a certain number of laps have been completed.
-
-**Tracing the route**: After the algorithm has finished, the optimal route can be found by
-backtracking the updated arrival
-times from the destination station to the start station.
-
-The **RAPTOR** algorithm is very efficient because it takes advantage of the fact that public
-transportation systems usually have regular schedules and that it is often faster to stay on
-the same line instead of switching. It is particularly well suited for real-time
-routing and planning in large transportation networks.
-
-## Flowchart
+The **RAPTOR** (Round-based Public Transit Routing) algorithm is a powerful algorithm to compute optimal routes in
+public transportation systems, especially for large and complex networks. Its primary objective is to efficiently
+determine the best route options based on factors like arrival time and number of transfers. RAPTOR is particularly
+useful for real-time journey planning due to its speed and adaptability.
+
+### Key Concepts of RAPTOR:
+
+- **Initialization**: The algorithm starts at the origin station and time specified by the user. Initially, it only
+  considers the starting station, marking it for the first round of exploration.
+
+- **Round-based Search**: The search process is broken into several rounds. In each round, the algorithm evaluates all
+  transport lines connected to the current list of stations. It computes the earliest possible arrival times at the
+  stations along each line and adds them to the list of stations for the next round.
+
+- **Updating Arrival Times**: As the algorithm processes new stations, it updates their earliest arrival times. If it
+  finds a route to a station that is faster than the previously known route, the arrival time and corresponding route
+  are updated.
+
+- **Termination Condition**: The algorithm stops when it reaches the destination station with the optimal arrival time
+  or when no further improvements can be made after a set number of rounds.
+
+- **Tracing the Optimal Route**: Once the algorithm concludes, the optimal route is reconstructed by tracing the updated
+  arrival times back from the destination to the start station.
+
+RAPTOR is highly efficient because it leverages the structure of public transit systems, where it is often faster to
+remain on the same line rather than switching between multiple lines. This characteristic makes RAPTOR particularly
```

Comment:
This sentence is not really correct and also not really inline with what the RAPTOR returns. At this point I would highlight that algorithm has the advantage of operating on the schedule information directly with no need to build extensive graphs incorporating the complexity of transfers.

Reply:
RAPTOR is highly efficient because it operates directly on schedule information without the need to build extensive
graphs that incorporate the complexity of transfers. This characteristic makes RAPTOR particularly suitable for
large-scale transportation networks where real-time routing is essential.

Reply:
I would go with the first suggestion, although the fact about accessibility and bike information is correct, it doesn't really fit without further explanation.